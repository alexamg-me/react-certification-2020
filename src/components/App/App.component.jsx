import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import AuthProvider from '../../providers/Auth';
import HomePage from '../../pages/Home';
import LoginPage from '../../pages/Login';
import NotFound from '../../pages/NotFound';
import SecretPage from '../../pages/Secret';
import Private from '../Private';
import Layout from '../Layout';
import VideoDetailPage from '../../pages/VideoDetail';
import FavoritePage from '../../pages/Favorite';

function App() {
  return (
    <BrowserRouter>
      <AuthProvider>
        <Layout>
          <Switch>
            <Route exact path="/">
              <HomePage />
            </Route>
            <Route exact path="/detail">
              <VideoDetailPage />
            </Route>
            <Route exact path="/favorites">
              <FavoritePage />
            </Route>
            <Route exact path="/favoritedetail">
              <VideoDetailPage favorite />
            </Route>
            <Route exact path="/login">
              <LoginPage />
            </Route>
            <Private exact path="/secret">
              <SecretPage />
            </Private>
            <Route path="*">
              <NotFound />
            </Route>
          </Switch>
        </Layout>
      </AuthProvider>
    </BrowserRouter>
  );
}

export default App;
