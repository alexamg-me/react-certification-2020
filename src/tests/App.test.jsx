import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from '../components/App/App.component';

describe(`Component: App`, () => {
  test(`App renders with default props`, () => {
    const component = TestRenderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
