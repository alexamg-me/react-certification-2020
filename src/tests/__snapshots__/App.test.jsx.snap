// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App App renders with default props 1`] = `
<main
  className="container"
>
  <section
    className="homepage"
  >
    <div
      className="sc-crrsfI qrNWt"
    >
      <div
        className="sc-dQppl ea-dLjL"
      >
        <div
          className="sc-hKgILt hpfQOr"
        >
          <div
            className="sc-eCssSg fSuOCU"
          >
            <div
              className="sc-jSgupP eKSphI"
            />
            <div
              className="sc-jSgupP eKSphI"
            />
            <div
              className="sc-jSgupP eKSphI"
            />
          </div>
        </div>
        <div
          className="sc-bdfBwQ ffLsOo"
        >
          <div
            className="sc-gsTCUz HFjtR"
          />
          <input
            className="sc-dlfnbm ifQijI"
            type="text"
          />
        </div>
      </div>
      <div
        className="sc-bqyKva duXVGw"
      >
        <div
          className="sc-gKsewC gcycVY"
        >
          <div
            className="sc-iBPRYJ gqNZGR"
          >
            <div
              className="sc-fubCfw gWoYKE"
            />
            <div
              className="sc-pFZIQ xJzqo"
            />
          </div>
          <div
            className="sc-jrAGrp bqZhEl"
          >
            Dark Mode
          </div>
        </div>
        <div
          className="sc-kEjbxe cpepa-d"
        >
          <div
            className="sc-iqHYGH fIAESH"
          />
        </div>
      </div>
    </div>
    <div
      className="sc-idOhPF dItPOG"
    >
      <div
        className="sc-fFubgz dvMlys"
      >
        <div
          className="sc-bkzZxe eROUxg"
        >
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Video Tour | Welcome to Wizeline Guadalajara
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/HYyRZiwBWc8/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline Guadalajara | Bringing Silicon Valley to Mexico
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/Po3VwR_NNGk/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline hace sentir a empleados como en casa
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/7PtYNO6g7eI/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              We Are Wizeline
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/YuW0CE_8i1I/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline Thrives in Mexico City
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/CHzlSGRvWPs/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/cjO2fJy8asM/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              A Day in the Life of an Engineering Manager at Wizeline
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/8bz9R61oY5o/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Silicon Valley en México
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/7dJFraOqcoQ/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Why Wizeline? featuring Juan Pablo Villa in Mexico City
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/w-Qwc_XJrWc/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline's 2020 Year in Review
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/rjir_cHTl5w/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Why Wizeline? featuring Hugo Lopez in Mexico City
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/DcFK1x3NHGY/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Why Wizeline? (We're Hiring in Mexico!)
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/3BzYWAqZgFw/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Why Wizeline? featuring Oswaldo Herrera in Mexico City
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/3KVFmT-Tp2w/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Caso de Éxito AWS: Wizeline [Spanish]
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/aKuPmY2m1Ro/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline's 2019 Year in Review
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/24sTHUyWhRM/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              [1 of 2] Wizeline CEO shares career lessons from Google
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/IxGc1gSqB3A/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline Data Engineering featuring Tania Reyes
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/NP1gAnbeNno/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Wizeline Querétaro | Mexico's New Knowledge Economy (We're hiring!)
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/F6Krwu6lUc8/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Miriam Godinez | Women in Leadership at Wizeline
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/RFq7gfvhtCk/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Welcome Back to Wizeline Vietnam | Extended Version
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/E1Vq_A3WKK8/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              How does Wizeline work?
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/ZmkslANDz0Q/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              12 Wishes from Wizeline | Happy Holidays 2019
            </p>
          </div>
          <div
            className="sc-kstrdz bgKXrU"
          >
            <img
              className="sc-fodVxV dDcGWn"
              src="https://i.ytimg.com/vi/Nss3EmTDD3s/hqdefault.jpg"
            />
            <p
              className="sc-hBEYos dgXxl"
            >
              Why Wizeline?
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-dIUggk bkuFeF"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        let me in →
      </a>
    </div>
  </section>
</main>
`;
